valve/addons/amxmodx/scripting/agmodx_llhl.sma

```diff
@@ -0,0 +1,430 @@
+    return HAM_IGNORED;
+}
+
+public Ham_Player_Spawn_Post(id) {
+    gIsAlive[id] = bool:is_user_alive(id);
+    gProbeCount[id] = gProbeSum[id] = 0;
+    return HAM_IGNORED;
+}
+
+public Ham_Player_Killed_Post(id) {
+    gIsAlive[id] = false;
+    return HAM_IGNORED;
+}
+
+// Create task before intermission because we can't run tasks when in intermission mode
+public agmodx_pre_intermission_mode() {
+    set_task(0.25, "taskShowVEngine", TASK_SHOWVENGINE, .flags = "b");
+    gActualServerFPS = gServerFPS;
+}
+
+public Ham_Game_End(id) {
+    // Don't calculate player fps when map is finished
+    remove_task(TASK_FPSLIMITER);
+
+    new players[MAX_PLAYERS], numPlayers;
+    get_players(players, numPlayers);
+    
+    new player;
+    for (new i; i < numPlayers; i++) {
+        player = players[i];
+        hl_strip_user_weapons(player);
+        client_cmd(player, "+showscores");
+    }
+}
+
+public FwStartFrame() {
+    static Float:gametime, Float:framesPer = 0.0;
+    static Float:tempFps;
+    
+    gametime = get_gametime();
+    
+    if(framesPer >= gametime) {
+        tempFps += 1.0;
+    } else {
+        framesPer = framesPer + 1.0;
+        gServerFPS = tempFps;
+        tempFps = 0.0;
+    }
+}
+
+public fmCmdStart_Pre(id, uc_handle) {
+    if (__is_user_alive(id)) {
+		static iMSec;
+		iMSec = get_uc(uc_handle, UC_Msec);
+
+		gProbeSum[id] += iMSec;
+		gProbeCount[id]++;
+
+		gLastProbeFPS[id] = precision_round(1 / (iMSec * 0.001), 0.1);
+	}
+    return PLUGIN_CONTINUE;
+}
+
+public taskMeasureMeanFPS() {
+    static players[MAX_PLAYERS], pCount, id;
+    get_players(players, pCount, "ch");
+    
+    for (new i = 0; i < pCount; i++) {
+        id = players[i];
+        if (gProbeCount[id] && !hl_get_user_spectator(id)) {
+            gMeanFPS[id] = precision_round(1 / ((gProbeSum[id] * 0.001) / gProbeCount[id]), 0.1);
+            gProbeCount[id] = gProbeSum[id] = 0;
+            if ((gMeanFPS[id] >= get_pcvar_float(gCvarFpsMax) + get_pcvar_float(gCvarFpsMarginError))) {
+                if (++gNumDetections[id] < get_pcvar_num(gCvarFpsMaxDetections)) {
+                    console_cmd(id, "^"FpS_MaX^" %d", get_pcvar_num(gCvarFpsMax));
+                    // For P47
+                    console_cmd(id, "^"FpS_ModeM^" %d", get_pcvar_num(gCvarFpsMax));
+                } else {
+                    static name[MAX_NAME_LENGTH];
+                    get_user_name(id, name, charsmax(name));
+                    server_cmd("kick #%d ^"%L^"", get_user_userid(id), id, "LLHL_FPSL_KICK", get_pcvar_num(gCvarFpsMax));
+                    client_print(0, print_chat, "%L", LANG_PLAYER, "LLHL_FPSL_KICK_MSG", name);
+                }
+            }
+        }
+    }
+}
+
+public taskShowVEngine() {
+    set_dhudmessage(0, 100, 200, -1.0, -0.125, 0, 0.0, 10.0, 0.2);
+    show_dhudmessage(0, "Llhl Mode vEngine^n----------------------^nServer fps: %.1f^nFilecheck: %s", gActualServerFPS, get_pcvar_num(gCvarCheckSoundFiles) ? "On" : "Off");
```

Comment:
I think it would be more better to capitalize the gamemode name. By the way, what's the meaning of vEngine? or is it just there to make it look cooler? 😄

Reply:
Done. I got the vEngine thing from AG's EHLL gamemode, I don't know exactly what it means either, but it looks good.
Do you want me to remove it or change it to another message?

Reply:
Nah, it's fine. Probably what you can do is add translation for that message too considering there's a multilingual file.